### ISSUE TITLE
Bug wit current AzureAdOidcConfiguration - it does not support OAuth 2.0 endpoint

### ISSUE THREAD

**Issue by mores** [pac4j/pac4j#1066]
DiscoveryURI is hardcoded to:
return "https://login.microsoftonline.com/"+tenant+"/.well-known/openid-configuration";

It should also support:
https://login.microsoftonline.com/tenantid/v2.0/.well-known/openid-configuration

**Comment by leleuj** [comment]
If you set the `tenant` property to "tenantid/v2.0", the second URL is supported, isn't it?

**Comment by mores** [comment]
I suppose you are correct....I don't like overloading the tenantid with a version. Other calls to AzureAd will require ONLY the tenantid ( future PRs ).

I am working on a PR to add a version attribute, but testing my app with v2.0 is not working as expected.
In the 2.2 branch I had been getting the error: "AADSTS70001: Application 'XXX-XXX-XXX-XXX' is not supported for this API version." I am not seeing that error anymore and wondering where it has gone to.

To summarize, I just am not sure that AzureAd v2.0 is 100% ready for release.

**Comment by leleuj** [comment]
Yes, it's a bit hacky. As we have an `AzureAdOidcConfiguration` object in the version 3.0.0, we could have maybe different configuration components for the different versions supported by AzureAD...

**Comment by leleuj** [comment]
@mores : so what do we do on this finally?

**Comment by leleuj** [comment]
Closing for now. Feel free to reopen with a contribution.
